### DISCUSSION TITLE
Switching Baudrate after inital handshake with power meter

### DISCUSSION THREAD

**Discussion by JYPDWhite** [arendst/Tasmota#19213] [Q&A]
Hi everyone, 

I'm using Tasmota on a ESP01 to communicate with my power meter via the optical interface (ISKA MT174).  I found this script online in the wiki 
```
>D
>B
->sensor53 r
>M 1
+1,3,o,0,300,STROM,1,100,2F3F210D0A
1,1-0:1.8.1255(@1,Total Consumed,KWh,Total_in,3
1,1-0:2.8.1255(@1,Total Delivered,KWh,Total_out,3
1,1-0:0.0.0*255(@#),Meter Number,,Meter_number,0
```
But with this script the communication is done with only 300 Baud. The meter can switch to 9600 Baud after the initial "handshake"  but I don't have a clue how to modify the script to accept the new Baudrate.
In the manual of the meter is described that the meter waits for 2 seconds after it sends / I S K 5 M T174 – “Program version” 
During this 2 seconds the client needs to answer with ACK 0 5 0 CR LF. to continue with 9600 Baud.

Does anyone have a tip for me how to achieve this?

Here is a Screenshot from the manual:
![grafik](https://github.com/arendst/Tasmota/assets/13840485/20333604-72e8-4b7f-a411-b336f8118504)

**Comment by 50l1d7r33** [comment]
I'm facing the same problem. With 300 Baud you don't get all information a special power consumption ist not provided. In the Documentation is written that you are able to change the Baud-Rate. There is also a Script that is not working 4 me. but I will share it anyway. Maybe some else facing my mistake:

```
>D
>B
->sensor53 r
>F
;count 100ms   
scnt+=1  
switch scnt  
case 6  
;set sml driver to 300 baud and send /?! as HEX to trigger the Meter   
res=sml(1 0 300)  
res=sml(1 1 "2F3F210D0A")  
;1800ms later \> Send ACK and ask for switching to 9600 baud  
case 18  
res=sml(1 1 "ACK 0 5 0 CR LF")  
;2000ms later \> Switching sml driver to 9600 baud    
case 20  
res=sml(1 0 9600)   
;Restart sequence after 50x100ms    
case 50  
; 5000ms later \> restart sequence    
scnt=0  
ends      
>M 1
+1,3,o,0, 9600,STROM,1,100
1,1-0:1.8.1*255(@1,Total Consumed,KWh,Total_in,3
1,1-0:1.8.0*255(@1,Total Delivered,KWh,Total_out,3
1,1-0:1.7.0*255(@1,Curent Power,KWh,power,3
1,1-0:0.0.0*255(@#),Meter Number,,Meter_number,0
#
```

**Reply by JYPDWhite** [reply]
Hi @50l1d7r33 I think your assumption is wrong. Switching the Baudrate will not provide you with the 1.7.0 value. I asked my energy meter provider and he told me that they configured the meter in a way that the 1.7.0 is not provided via the IR interface because of data privacy considerations.

**Reply by 50l1d7r33** [reply]
Hi @JYPDWhite , thank you 4 your answer, I have also already consulted my energy supplier and the company that maintains the electricity meters, unfortunately without getting such a statement. A pin is also not set and I do not see any other way to display the value. Too bad, it would even be one of the most important to me. However, I have not managed to change the baud rate, even if it would be superfluous. Strange is that I can display the value for 1.7.0 in the menu of the counter. What would preventing the readout over IR still use there? Anyway, I guess we'll never know.

**Reply by JYPDWhite** [reply]
Yes its strange that displaying the 1.7.0 value in the display is no privacy issue but over the IR interface it is .....
For you information I've got the information that if I request a new meter with the reason of "ablesbarkeit" (readability) then I don't get charged for the new meter.
